src/Sasl/Plain.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/* vim: set expandtab tabstop=4 shiftwidth=4 softtabstop=4 foldmethod=marker: */
+// +---------------------------------------------------------------------------
+// | SWAN [ $_SWANBR_SLOGAN_$ ]
+// +---------------------------------------------------------------------------
+// | Copyright $_SWANBR_COPYRIGHT_$
+// +---------------------------------------------------------------------------
+// | Version  $_SWANBR_VERSION_$
+// +---------------------------------------------------------------------------
+// | Licensed ( $_SWANBR_LICENSED_URL_$ )
+// +---------------------------------------------------------------------------
+// | $_SWANBR_WEB_DOMAIN_$
+// +---------------------------------------------------------------------------
+
+namespace Kafka\Sasl;
+
+use Kafka\CommonSocket;
+use Kafka\SaslMechanism;
+use Kafka\Exception;
+
+/**
++------------------------------------------------------------------------------
+* Kafka sasl provider for plain mechanism
++------------------------------------------------------------------------------
+*
+* @package
+* @version $_SWANBR_VERSION_$
+* @copyright Copyleft
+* @author $_SWANBR_AUTHOR_$
++------------------------------------------------------------------------------
+*/
+
+class Plain extends Mechanism implements SaslMechanism
+{
+    // {{{ consts
+    
+    private const MECHANISM_NAME = "PLAIN";
+
+    // }}}
+    // {{{ members
+
+    private $username;
+
+    private $password;
+
+    // }}}
+    // {{{ functions
+    // {{{ public function __construct()
+    
+    /**
+     *
+     * __construct
+     *
+     * @access public
+     * @return void
+     */
+    public function __construct(array $options)
```

Comment:
`public function __construct(string $username, string $password)` is more appropriate here because:

- simplicity: no need for check for keys in the array
- explicitness: the required arguments are quite clear for the users
- performance: PHP arrays are very flexible but they perform terribly (ok, it's micro optimisation)